rust/src/lib/unit_tests/net_state.rs

```diff
@@ -23,3 +28,46 @@ fn test_invalid_top_type() {
 
     assert!(result.is_err());
 }
+
+#[test]
+fn test_network_state_debug() {
+    let mut network_state = NetworkState {
+        hostname: Some(HostNameState::default()), // Use Option<HostNameState>
+        dns: None,
+        rules: RouteRules::default(),
+        routes: Routes::default(),
+        interfaces: Interfaces::default(),
+        ovsdb: None,
+        timeout: Some(60),
+        no_verify: true,
+        no_commit: false,
+        ovn: OvnConfiguration::default(),
+        include_secrets: true,
+        include_status_data: false,
+        running_config_only: true,
+        memory_only: false,
+        kernel_only: false,
+    };
+
+    network_state.hide_secrets();
```

Comment:
Please remove this line. Even without explicitly call of `hide_secrets()`, debug output of NetworkState should never reveal secrets.